model/generate.go

```diff
@@ -93,6 +93,18 @@ func MergeGeneratedProjects(projects []GeneratedProject) *GeneratedProject {
 	return g
 }
 
+// ParseProjectFromJSON returns a GeneratedTasks type from JSON. We use the
+// YAML parser instead of the JSON parser because the JSON parser will not
+// properly unmarshal into a struct with multiple fields as options, like the YAMLCommandSet.
+func ParseProjectFromJSONString(data string) (GeneratedProject, error) {
+	g := GeneratedProject{}
```

Comment:
is there a good reason that this function doesn't return a pointer? I think returning concrete types is probably wrong most of the time.

Reply:
Returning a pointer makes is awkward for the caller. I'd prefer not to refactor in this ticket.

Reply:
I feel like the pointer+error paradigm is pretty well established, and getting a non-nill but zero value is confusing?

Reply:
The existing ParseProjectFromJson function already has this function signature though, this function is just to handle a different case.